test/common-tests/request-field-validation-api-tests/string-field-validation-api-tests.ts

```diff
@@ -13,13 +17,20 @@ interface StringFieldValidationApiTestOptions<RequestBodyItem, RequestBodyItemKe
 export function withStringFieldValidationApiTests<RequestBodyItem, RequestBodyItemKey extends keyof RequestBodyItem>({
   fieldName: fieldNameSymbol,
   length,
+  minLength,
+  maxLength,
   required,
+  pattern,
   generateFieldValueOfLength,
+  generateFieldValueThatDoesNotMatchRegex,
   validRequestBody,
   makeRequest,
   givenAnyRequestBodyWouldSucceed,
 }: StringFieldValidationApiTestOptions<RequestBodyItem, RequestBodyItemKey>): void {
   const fieldName = fieldNameSymbol.toString();
+  minLength = minLength ?? length;
+  maxLength = maxLength ?? length;
```

Comment:
Minor point, but I'd expect `length` to take priority over `minLength` and `maxLength` if `length` is set, so
```
minLength = length ?? minLength;
maxLength = length ?? maxLength;
```

Reply:
Similarly, I changed the code above to match your suggestion and also changed the type to:
```
type StringFieldValidationApiTestOptions<RequestBodyItem, RequestBodyItemKey extends keyof RequestBodyItem> =
  | {
      fieldName: RequestBodyItemKey;
      length: number;
      minLength?: undefined;
      maxLength?: undefined;
      required?: boolean;
      pattern?: string;
      generateFieldValueOfLength: (length: number) => RequestBodyItem[RequestBodyItemKey];
      generateFieldValueThatDoesNotMatchRegex?: () => RequestBodyItem[RequestBodyItemKey];
      validRequestBody: RequestBodyItem[];
      makeRequest: (body: unknown[]) => request.Test;
      givenAnyRequestBodyWouldSucceed: () => void;
    }
  | {
      fieldName: RequestBodyItemKey;
      length?: undefined;
      minLength: number;
      maxLength: number;
      required?: boolean;
      pattern?: string;
      generateFieldValueOfLength: (length: number) => RequestBodyItem[RequestBodyItemKey];
      generateFieldValueThatDoesNotMatchRegex?: () => RequestBodyItem[RequestBodyItemKey];
      validRequestBody: RequestBodyItem[];
      makeRequest: (body: unknown[]) => request.Test;
      givenAnyRequestBodyWouldSucceed: () => void;
    };
```
so that the developer has to specify either `length` or (`minLength` and `maxLength`)

Reply:
As above, apologies, I've just rebased onto main and seen that a function has been added that throws errors if both `length` and (`minLength` or `maxLength`) are specified